src/mono/mono/mini/method-to-ir.c

```diff
@@ -7565,12 +7567,25 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 							}
 						} else {
 							// For structs that cannot be lowered, we change the argument to byref type
-							ptype = mono_class_get_byref_type (mono_defaults.typed_reference_class);
-							// Load the address of the struct
+							if (gklass && (gklass->container_class == swift_self_t))
+								ptype = mono_class_get_byref_type (swift_self);
+							else
+								ptype = mono_class_get_byref_type (mono_defaults.int_class);
```

Comment:
Why is it `mono_class_get_byref_type (mono_defaults.int_class)` instead of `mono_class_get_byref_type (mono_defaults.typed_reference_class)` or `mono_class_get_byref_type (klass);` ?

Reply:
https://github.com/dotnet/runtime/pull/104171#discussion_r1659241578

Reply:
@jkurdek `TypedReference&` is kind of weird since it's a ref of a byreflike struct.  I don't remember if we support that in .NET yet.  it might make Mono do something weird.

`mono_class_get_byref_type (klass)` might be ok.

@kotlarmilos I think it's probably worth adding some comment about the placeholder type here - it's arbitrary and not really meant to exactly capture the exact type - all we care about is that it's some kind of pointer-like argument.